File: job-hunt-frontend/job-hunt-ui/src/services/PositionService.js
import axios from "axios";
import { getToken } from "./AuthService";

const REST_API_POSITION_BASE_URL = 'http://localhost:8080/api/positions';

// Add a request interceptor
axios.interceptors.request.use(function (config) {
    config.headers['Authorization'] = getToken();
    return config;
  }, function (error) {
    return Promise.reject(error);
  });


export const getAllPositions = () => axios.get(REST_API_POSITION_BASE_URL); // GET ALL

export const addPosition = (position) => axios.post(REST_API_POSITION_BASE_URL, position); // ADD 

export const deletePosition = (id) => axios.delete(REST_API_POSITION_BASE_URL + '/' + id); // DELETE

export const updatePosition = (id, position) => axios.put(REST_API_POSITION_BASE_URL + '/' + id, position); // UPDATE

export const getPosition = (id) => axios.get(REST_API_POSITION_BASE_URL + '/' + id); // GET

File: job-hunt/src/main/java/ykpsph/jobhunt/service/PositionService.java
package ykpsph.jobhunt.service;


import ykpsph.jobhunt.dto.PositionDTO;

import java.util.List;

public interface PositionService {
    PositionDTO addPosition(PositionDTO positionDTO); // ADD
    List<PositionDTO> getAllPositions(); // GET ALL
    PositionDTO getPosition(Long id); // GET
    PositionDTO updatePosition(PositionDTO positionDTO, Long id); // UPDATE
    void deletePosition(Long id); // DELETE

    // nothing to change here

}
